pkg/doc/jose/jwe.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package jose
+
+// LegacyEncodingType is the `typ` string identifier in a message that identifies the format as being legacy
+const LegacyEncodingType string = "JWM/1.0"
+
+// LegacyEnvelope is the full payload envelope for the JSON message
+type LegacyEnvelope struct {
+	Protected  string `json:"protected,omitempty"`
+	IV         string `json:"iv,omitempty"`
+	CipherText string `json:"ciphertext,omitempty"`
+	Tag        string `json:"tag,omitempty"`
+}
+
+// LegacyProtected is the protected header of the JSON envelope
+type LegacyProtected struct {
+	Enc        string            `json:"enc,omitempty"`
+	Typ        string            `json:"typ,omitempty"`
+	Alg        string            `json:"alg,omitempty"`
+	Recipients []LegacyRecipient `json:"recipients,omitempty"`
+}
+
+// LegacyRecipient holds the data for a recipient in the envelope header
+type LegacyRecipient struct {
+	EncryptedKey string                `json:"encrypted_key,omitempty"`
+	Header       LegacyRecipientHeader `json:"header,omitempty"`
+}
+
+// LegacyRecipientHeader holds the header data for a recipient
+type LegacyRecipientHeader struct {
+	KID    string `json:"kid,omitempty"`
+	Sender string `json:"sender,omitempty"`
+	IV     string `json:"iv,omitempty"`
+}
+
+// ToDo/ Code review question... can this just be called a "standard JWE" struct?
+// ToDo (continued) Rather than an "Aries Encryption envelope"?
+// ToDo (continued) It looks like a standard JWE to me, unless I'm mistaken...
```

Comment:
@Baha-sk

Reply:
I wouldn't move old formats to keep all legacy structs and code in one place. It will make it easier to decommission later on.

Reply:
(done!... or "not done", technically)